packages/protocol/test/common/recoverFunds.ts

```diff
@@ -0,0 +1,65 @@
+// Note: this test is testing the recover-funds script and not a particular smart contract.
+
+import { recoverFunds } from '@celo/protocol/lib/recover-funds'
+import { CeloContractName } from '@celo/protocol/lib/registry-utils'
+import { expectBigNumberInRange } from '@celo/protocol/lib/test-utils'
+import { BigNumber } from 'bignumber.js'
+import {
+  FreezerContract,
+  GetSetV0Instance,
+  GoldTokenContract,
+  ProxyInstance,
+  RegistryContract,
+} from 'types'
+
+const GetSetV0: Truffle.Contract<GetSetV0Instance> = artifacts.require('GetSetV0')
+const Proxy: Truffle.Contract<ProxyInstance> = artifacts.require('Proxy')
+
+contract('Proxy', (accounts: string[]) => {
+  let proxy: ProxyInstance
+  let getSet: GetSetV0Instance
+
+  const owner = accounts[0]
+
+  beforeEach(async () => {
+    proxy = await Proxy.new({ from: owner })
+    getSet = await GetSetV0.new({ from: owner })
+  })
+
+  describe('fallback', () => {
+    beforeEach(async () => {
+      await proxy._setImplementation(getSet.address)
+    })
+
+    it('recovers funds from an incorrectly intialized implementation', async () => {
```

Comment:
a bit confused about this test, as it still uses truffle instead of foundry

Reply:
This one is actually a test in proxy that I renamed, the reason it still stands it's because it's not testing the contract but rather the ts function `recoverFunds `, which we have no way to test until we figure out the anvil integration.